emu/src/render/ppu.rs

```diff
@@ -31,11 +35,9 @@ impl PixelProcessUnit {
     }
 
     pub fn render(&self) {
-        let memory = self.internal_memory.lock().unwrap();
-        let mut gba_lcd = self.gba_lcd.lock().unwrap();
-
         #[allow(unused_assignments)]
-        let mut bg_mode = memory.lcd_registers.get_bg_mode();
+        let mut bg_mode =
+            acquire_lock!(self.internal_memory, memory => { memory.lcd_registers.get_bg_mode() });
```

Comment:
I think we could only lock once but I guess it's not a big deal

Reply:
Yeah. I had the same idea, but `get_color_from_full_palette()` in `BG_MODE_4` uses the same lock.